src/Raven.Client/Http/NodeSelector.cs

```diff
@@ -28,6 +29,43 @@ public NodeSelectorState(Topology topology)
                 UnlikelyEveryoneFaultedChoiceIndex = 0;
             }
 
+            public NodeSelectorState(Topology topology, NodeSelectorState prevState)
+            {
+                Topology = topology;
+                Nodes = topology.Nodes;
+                Failures = new int[topology.Nodes.Count];
+                FastestRecords = new int[topology.Nodes.Count];
+                UnlikelyEveryoneFaultedChoiceIndex = 0;
+                
+                var fastestNode = prevState.Nodes.ElementAt(prevState.Fastest);
+                int index = 0;
+                foreach (var node in topology.Nodes)
+                {
+                    if (node.ClusterTag == fastestNode.ClusterTag)
+                    {
+                        Fastest = index;
+                        break;
+                    }
+
+                    index++;
+                }
+                
+                // fastest node was not found in the new topology. enable speed tests
+                if (index >= topology.Nodes.Count)
+                {
+                    SpeedTestMode = 2;
+                }
+                else
+                {
+                    // we might be in the process of finding fastest node when we reorder the nodes, we don't want the tests to stop until we reach 10
+                    // otherwise, we want to stop the tests and they may be scheduled later on relevant topology change
+                    if (Fastest < prevState.FastestRecords.Length && prevState.FastestRecords[Fastest] < 10)
```

Comment:
maybe I'm missing something here... but we aren't copying the content of the old `prevState.FastestRecords` to the new one... 
which might lead, if the topology was changed _twice_ we will end up in doing a speed test even when this not necessary, right?

Reply:
Yes, only if the topology changes happened at a time when we still haven't finished the 10 requests. Is it worth it to add the complexity to save a few extra requests which will only happen in a pretty narrow window?

Reply:
Let's assume the original one had 10 in the FastestRecords...
now we update the topology and have 0 and we do not trigger the fast test phase
but now further topology update will trigger the fast test phase